packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol

```diff
@@ -275,6 +296,88 @@ contract L1CrossDomainMessenger_Test is CommonTest {
         );
     }
 
+    /// @notice Tests that the relayMessage function on L2 will always succeed for any potential
+    ///         message, regardless of the size of the message, the minimum gas limit, or the
+    ///         amount of gas used by the target contract.
+    function testFuzz_relayMessage_baseGasSufficient_succeeds(
+        uint24 _messageLength,
+        uint32 _minGasLimit,
+        uint32 _gasToUse
+    )
+        external
+    {
+        // Skip if this is a fork test, won't work.
+        skipIfForkTest("L2CrossDomainMessenger doesn't exist on L1 in forked test");
+
+        // Using smaller uint so the fuzzer doesn't give as many massive values that get bounded.
+        // TODO: Known issue, messages above 34k can actually OOG on the receiving side if the
+        // target uses all available gas. Can be resolved by capping data sizes on the XDM or by
+        // increasing the amount of available relay gas to ~100k. If increasing relay gas, should
+        // have the relay gas only increase when the calldata size is large to avoid disrupting
+        // in-flight L2 -> L1 messages.
+        _messageLength = uint24(bound(_messageLength, 0, 34_000));
+
+        // Need more than 500 since GasBurner requires it.
+        // It's ok to try to use more than minGasLimit since the L2CrossDomainMessenger should
+        // catch the revert and store the message hash in the failedMessages mapping.
+        _gasToUse = uint32(bound(_gasToUse, 500, type(uint32).max));
+
+        // Generate random bytes, more useful than having a _message parameter.
+        bytes memory _message = vm.randomBytes(_messageLength);
```

Comment:
This is fine, just curious why this way is more useful?

Reply:
Easier to bound the length this way I'm guessing than to get super large calldata params, which might not even be that long?

Reply:
Yeah, the super large calldata parameters end up not actually being long enough to test this.